client.go

```diff
@@ -507,9 +514,22 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 			return nil, errMissingDatabasePoolWithQueues
 		}
 
-		client.completer = jobcompleter.NewBatchCompleter(archetype, driver.GetExecutor(), nil)
+		// The batch completer is much faster when completing many jobs, but it
+		// pauses before completing jobs, waiting for more potential completions
+		// to come in which can be added to the batch. This is good for live
+		// environments but not as good for tests, where it adds some delay to
+		// every test case waiting for a job completion. With TestOnly on, use
+		// the async completer instead, which despite its name, starts
+		// incoming completions immediately by doing them in a goroutine.
+		if config.TestOnly {
+			client.completer = jobcompleter.NewAsyncCompleter(archetype, driver.GetExecutor(), nil)
+		} else {
+			client.completer = jobcompleter.NewBatchCompleter(archetype, driver.GetExecutor(), nil)
+		}
+		client.services = append(client.services, client.completer)
+
 		client.subscriptionManager = newSubscriptionManager(archetype, nil)
-		client.services = append(client.services, client.completer, client.subscriptionManager)
+		client.services = append(client.services, client.subscriptionManager)
```

Comment:
It would be cleaner to delete the first append and merge it into the second with `append(client.services, client.completer, client.subscriptionManager)`. This potentially also avoids an unnecessary allocation as append only needs to allocate more memory once instead of potentially twice.

Reply:
Realistically, if Go does need to grow the slice, it'll 2x the underlying size, so it's not going to make any difference. I'd take the small readability advantage instead of trying to micro-optimize here.